src/Tracing/DynamicSamplingContext.php

```diff
@@ -0,0 +1,205 @@
+ *
+ * @see https://develop.sentry.dev/sdk/performance/dynamic-sampling-context/
+ */
+final class DynamicSamplingContext
+{
+    public const SENTRY_ENTRY_PREFIX = 'sentry-';
+
+    /**
+     * The dsc entries.
+     *
+     * @var array<string, string>
+     */
+    private $entries;
+
+    /**
+     * Indicates if the dsc is mutalbe or immutable.
+     *
+     * @var bool
+     */
+    private $isFrozen = false;
+
+    /**
+     * Construct a new dsc object.
+     */
+    private function __construct()
+    {
+        $this->entries = [];
+    }
+
+    /**
+     * Set a new key value pair on the dsc.
+     *
+     * @param string $key   the list member key
+     * @param string $value the list member value
+     */
+    public function set(string $key, string $value): void
+    {
+        if ($this->isFrozen) {
+            return;
+        }
+
+        $this->entries[$key] = $value;
+    }
+
+    /**
+     * Check if a key value pair is set on the dsc.
+     *
+     * @param string $key the list member key
+     */
+    public function has(string $key): bool
+    {
+        return isset($this->entries[$key]);
+    }
+
+    /**
+     * Get a value from the dsc.
+     *
+     * @param string      $key     the list member key
+     * @param string|null $default the default value to return if no value exists
+     */
+    public function get(string $key, ?string $default = null): ?string
+    {
+        return $this->entries[$key] ?? $default;
+    }
+
+    /**
+     * Mark the dsc as frozen.
+     */
+    public function freeze(): void
+    {
+        $this->isFrozen = true;
+    }
+
+    /**
+     * Indicates that the dsc is frozen and cannot be mutated.
+     */
+    public function isFrozen(): bool
+    {
+        return $this->isFrozen;
+    }
+
+    /**
+     * Check if there are any entries set.
+     */
+    public function hasEntries(): bool
+    {
+        return !empty($this->entries);
+    }
+
+    /**
+     * Gets the dsc entries.
+     *
+     * @return array<string, string>
+     */
+    public function getEntries(): array
+    {
+        return $this->entries;
+    }
+
+    /**
+     * Parse the baggage header.
+     *
+     * @param string $header the baggage header contents
+     */
+    public static function fromHeader(string $header): self
+    {
+        $dsc = new self();
+
+        foreach (explode(',', $header) as $listMember) {
+            if (empty(trim($listMember))) {
+                continue;
+            }
+
+            $keyValueAndProperties = explode(';', $listMember, 2);
+
+            $keyValue = trim($keyValueAndProperties[0]);
+
+            if (empty($keyValue) || !str_contains($keyValue, '=')) {
+                continue;
+            }
+
+            [$key, $value] = explode('=', $keyValue, 2);
+
+            if (0 === strpos($key, self::SENTRY_ENTRY_PREFIX)) {
+                $dsc->set(rawurldecode(str_replace(self::SENTRY_ENTRY_PREFIX, '', $key)), rawurldecode($value));
```

Comment:
Shouldn't we trim just the prefix and not all occurrences of this constant? I would have expected using `substr` to play safe